packages/scope-manager/src/referencer/Referencer.ts

```diff
@@ -752,8 +752,19 @@ class Referencer extends Visitor {
   }
 
   protected UpdateExpression(node: TSESTree.UpdateExpression): void {
-    if (PatternVisitor.isPattern(node.argument)) {
-      this.visitPattern(node.argument, pattern => {
+    let argument = node.argument;
```

Comment:
This could be extracted into a function (as it's the same as lines 328-338), but I have no idea how to name it.

Reply:
Good callout. Maybe... `visitExpressionLeft`? `visitExpressionLeftBypassingAssertion`? I'd put it at the end of `// Visit helpers` maybe.

Reply:
Coming back to this, I'll go with `visitExpressionTarget`. Only the other one of the two is about the _"left"_.